pkg/ansible/controller/reconcile.go

```diff
@@ -225,27 +223,21 @@ func (r *AnsibleOperatorReconciler) Reconcile(request reconcile.Request) (reconc
 		}
 	}
 	if r.ManageStatus {
-		err = r.markDone(u, request.NamespacedName, statusEvent, failureMessages)
-		if err != nil {
-			logger.Error(err, "Failed to mark status done")
-		}
+		merr := r.markDone(u, request.NamespacedName, statusEvent, failureMessages)
+		checkAPIErr(merr, logger)
 	}
-	return reconcileResult, err
+	return reconcileResult, nil
```

Comment:
Can we ensure that all errors above have been checked and the `reconcileResult, err` used in these scenarios?

Reply:
`err` is checked everywhere it is returned in this function. I'd consider returning a non-`nil` error here to be a bug, since it should've been checked and handled by the caller.

Reply:
I agree the nil is better, just a point of attention for we check and ensure that all errors have been returned above.